numba/cuda/cudadrv/devicearray.py

```diff
@@ -331,6 +331,13 @@ def view(self, dtype):
             gpu_data=self.gpu_data,
             )
 
+    @property
+    def nbytes(self):
+        if hasattr(self, 'alloc_size'):
+            return self.alloc_size
```

Comment:
To match numpy behavior, I don't think we can use the `.alloc_size`, which is reporting the memory consumption of the allocation.  For instance, 
```python
In [28]: np.arange(5)[::2].nbytes
Out[28]: 24

In [29]: cuda.to_device(np.arange(5))[::2].alloc_size
Out[29]: 40
```
gives different answer because `.nbytes` is only giving the size used by the elements in the array.

For broadcasted array, numpy's `.nbytes` will give a bigger number than what is actually allocated in memory.

Reply:
Ah, there is overhead in the allocation!  Making changes now